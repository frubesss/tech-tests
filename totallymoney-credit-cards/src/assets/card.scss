.card-wrapper {

  .card {
    background-color: #F9F9F9;
    padding: 20px;
    border: 1px solid #5b6b79;
    border-radius: 25px;
    width: 100%;
    max-width: 400px;
    margin-bottom: 10px;

    .title {
      font-size: 2rem;
      line-height: 3.51rem;
      font-weight: 700;
      text-align: center;
      margin-bottom: 10px;
    }

    p {
      margin: auto;
      font-size: 15px;
      text-align: left;
    }

  }

}

