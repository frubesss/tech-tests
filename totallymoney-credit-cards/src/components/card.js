import React from 'react';
import '../assets/card.scss'
import PropTypes from "prop-types";

export const Card = ({creditCards}) => {

    return (
        <div>
            {creditCards &&
            <div>
                {creditCards.map((card) => {
                        return <div className='card-wrapper' key={card.uuid} >
                            <div className='card'>
                                <p className="title">{card.name}</p>
                                <p className="apr">Apr: <strong>{card.apr}%</strong></p>
                                <p className="b-t">Balance Transfer Offer
                                    Duration: <strong>{card.balanceTransferDuration} Months</strong></p>
                                <p className="purchase">Purchase Offer
                                    Duration: <strong>{card.purchaseOfferDuration} Months</strong></p>
                                <p className="credit">Credit Available: <strong>Â£{card.credit}</strong></p>
                            </div>
                        </div>
                    }
                )}
            </div>
            }
        </div>

    );

};

Card.propTypes = {
    creditCards: PropTypes.array.isRequired
};
