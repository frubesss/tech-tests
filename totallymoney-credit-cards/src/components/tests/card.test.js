import React from 'react';
import renderer from 'react-test-renderer'
import {shallow} from 'enzyme';
import {Card} from "../card";

describe('Card Component', () => {

    let wrapper;

    const mockedTwoCreditCards = [
        {
            "uuid": "1232323",
            "name": "Student Life Card",
            "apr": "18.9",
            "balanceTransferDuration": "0",
            "purchaseOfferDuration": "6",
            "credit": "1200"
        },
        {
            "uuid": "55667",
            "name": "Anywhere Card",
            "apr": "33.9",
            "balanceTransferDuration": "0",
            "purchaseOfferDuration": "0",
            "credit": "300"
        }
    ];

    const mockedOneCreditCards = [
        {
            "uuid": "1232323",
            "name": "Student Life Card",
            "apr": "18.9",
            "balanceTransferDuration": "0",
            "purchaseOfferDuration": "6",
            "credit": "1200"
        }
    ];

    beforeEach(() => {
        wrapper = shallow(<Card creditCards={mockedTwoCreditCards}/>);
    });

    it('Snapshot Test', () => {
        const tree = renderer.create(wrapper).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('should render 2 credit cards when customer is eligible', () => {
        expect(wrapper.find('.card')).toHaveLength(2);
    });

    it('should render 1 credit cards when customer is eligible', () => {
        wrapper = shallow(<Card creditCards={mockedOneCreditCards}/>);
        expect(wrapper.find('.card')).toHaveLength(1);
    });

});
