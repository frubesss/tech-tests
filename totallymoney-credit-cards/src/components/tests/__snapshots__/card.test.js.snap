// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card Component Snapshot Test 1`] = `
<div>
  <div>
    <div
      className="card-wrapper"
    >
      <div
        className="card"
      >
        <p
          className="title"
        >
          Student Life Card
        </p>
        <p
          className="apr"
        >
          Apr: 
          <strong>
            18.9
            %
          </strong>
        </p>
        <p
          className="b-t"
        >
          Balance Transfer Offer Duration: 
          <strong>
            0
             Months
          </strong>
        </p>
        <p
          className="purchase"
        >
          Purchase Offer Duration: 
          <strong>
            6
             Months
          </strong>
        </p>
        <p
          className="credit"
        >
          Credit Available: 
          <strong>
            £
            1200
          </strong>
        </p>
      </div>
    </div>
    <div
      className="card-wrapper"
    >
      <div
        className="card"
      >
        <p
          className="title"
        >
          Anywhere Card
        </p>
        <p
          className="apr"
        >
          Apr: 
          <strong>
            33.9
            %
          </strong>
        </p>
        <p
          className="b-t"
        >
          Balance Transfer Offer Duration: 
          <strong>
            0
             Months
          </strong>
        </p>
        <p
          className="purchase"
        >
          Purchase Offer Duration: 
          <strong>
            0
             Months
          </strong>
        </p>
        <p
          className="credit"
        >
          Credit Available: 
          <strong>
            £
            300
          </strong>
        </p>
      </div>
    </div>
  </div>
</div>
`;
